nsqd/logger.go

```diff
@@ -18,3 +18,9 @@ func (n *NSQD) logf(level lg.LogLevel, f string, args ...interface{}) {
 	opts := n.getOpts()
 	lg.Logf(opts.Logger, opts.logLevel, level, f, args...)
 }
+
+// would like to expose logf to the contrib modules so that contrib can share the
+// configuration end user specifies on CLI
+func (n *NSQD) Logf(level lg.LogLevel, f string, args ...interface{}) {
```

Comment:
A possibly cleaner alternative is to pass the `logf` bound-method reference to `NewNSQDAddons()` or `NSQDAddons.Start()`, which could take a `internal.lg.AppLogFunc` type. For example see https://github.com/nsqio/nsq/blob/844c6a08f440d14b090bed6fa15722c327ebdd3a/nsqd/nsqd.go#L231 and https://github.com/nsqio/nsq/blob/844c6a08f440d14b090bed6fa15722c327ebdd3a/internal/protocol/tcp_server.go#L15